canarytokens/aws_infra/data_generation.py

```diff
@@ -0,0 +1,314 @@
+    return bool(re.fullmatch(_DYNAMO_DB_TABLE_NAME_REGEX, name))
+
+
+def _validate_ssm_parameter_name(name: str) -> bool:
+    # path segments of a-z A-Z 0-9 _ . - separated by "/"; no leading "aws" or "ssm"
+    if not name or len(name) > 2048:
+        return False
+    segments = name.split("/")
+    if segments[0].lower() in ("aws", "ssm"):
+        return False
+    for seg in segments:
+        if not seg:
+            continue  # skip empty segments
+        if not re.fullmatch(_SSM_PARAMETER_NAME_REGEX, seg):
+            return False
+    return True
+
+
+def _validate_sqs_name(name: str) -> bool:
+    # 1-80 chars, A-Z a-z 0-9 _ - ;
+    return bool(re.fullmatch(_SQS_QUEUE_NAME_REGEX, name))
+
+
+def _validate_secrets_manager_name(name: str) -> bool:
+    # 1-512 chars, A-Z a-z 0-9 / _ + = . @ - .
+    return bool(re.fullmatch(_SECRETS_MANAGER_NAME_REGEX, name))
+
+
+_VALIDATORS = {
+    AWSInfraAssetType.S3_BUCKET: _validate_s3_name,
+    AWSInfraAssetType.DYNAMO_DB_TABLE: _validate_dynamodb_name,
+    AWSInfraAssetType.SSM_PARAMETER: _validate_ssm_parameter_name,
+    AWSInfraAssetType.SQS_QUEUE: _validate_sqs_name,
+    AWSInfraAssetType.SECRETS_MANAGER_SECRET: _validate_secrets_manager_name,
+}
+
+
+async def _validate_name(asset_type: AWSInfraAssetType, name: str) -> bool:
+    """
+    Validate a name against the rules for the specified AWS asset type.
+    """
+    validator = _VALIDATORS.get(asset_type)
+    if not validator:
+        raise ValueError(f"No validator found for asset type {asset_type.name}.")
+
+    # Handle async validators (like S3) vs sync validators
+    if asset_type == AWSInfraAssetType.S3_BUCKET:
+        return await validator(name)
+    else:
+        return validator(name)
+
+
+def _augment_name(asset_type: AWSInfraAssetType, name: str) -> str:
+    """
+    Augment a name with a random suffix to ensure uniqueness.
+    """
+    if asset_type == AWSInfraAssetType.S3_BUCKET:
+        max_length = 63
+        min_random_suffix_length = 3
+        name = f"{name}-{random_string(random.randint(min_random_suffix_length, max_length // 3), lower_case_only=True)}"
+        if len(name) > max_length:
+            name = name[:max_length]
+
+    return name
+
+
+async def _finalize_list(
+    asset_type: AWSInfraAssetType,
+    inventory: list[str],
+    suggested_names: list[str],
+    count: int,
+) -> list[str]:
+    """
+    Get a list of validated names from the provided inventory.
+    """
+    validated_names = []
+    for name in suggested_names:
+        name = _augment_name(asset_type, name)
+        is_valid = await _validate_name(asset_type, name)
+        similarity_score = max(
+            (SequenceMatcher(None, name, existing).ratio() for existing in inventory),
+            default=0,
+        )
+        if is_valid and similarity_score < _SIMILARITY_THRESHOLD:
+            validated_names.append(name)
+
+    return validated_names[:count] if len(validated_names) >= count else validated_names
```

Comment:
You can count on Python to do the right thing here:
```suggestion
    return validated_names[:count]
```